sql/core/src/main/scala/org/apache/spark/sql/expressions/Aggregator.scala

```diff
@@ -58,6 +58,11 @@ abstract class Aggregator[-A, B, C] {
   def reduce(b: B, a: A): B
 
   /**
+   * Merge two intermediate values
+   */
+  def merge(b1: B, b2: B): B
```

Comment:
@mateiz JFYI, I missed adding this in the first iteration, but I think we need it if we are going to do multilevel aggregation.